articles/remapping.md

```diff
@@ -0,0 +1,77 @@
+### Remapping Names
+#### ROS 1
+In ROS 1 [Names](http://wiki.ros.org/Names) can be remapped to other names by passing [remapping arguments](http://wiki.ros.org/Remapping%20Arguments) via the command line. `roscpp` additionally allows remaps to be passed via Init as strings in STL containers. A ROS1 remap consists of two names: one that should be replaced with another.
+
+In ROS1 remapping works with FQN. When a name is to be checked for remappings, it is first expanded to a FQN. Both parts of the remapping rule are expanded to a FQN. Then a name is changed to the remap's second name only if it exactly matches the first name.
```

Comment:
I'm not completely certain this is how it works, though it might be correct. In fact it doesn't matter if you expand them first or not, I think. For example, if you create a topic called `foo`, and then you pass the remapping `foo:=bar` and you're in a namespace `/baz/`, then if you replace before expanding to a FQN you'd replace `foo` with `bar` and then expand that to `/baz/bar` (correct answer) but if you expand the replacements first you get the same result, i.e. `foo` becomes `/baz/foo` and the remapping becomes `/baz/foo:=/baz/bar` and then you replace it to get the correct answer again. So I think it's potentially unimportant when the remapping happens, at least in this case, but it would be better if you could cite this by linking to ROS 1 docs or even to the code which does this in `roscpp`.

Also, a nitpick, you should use the full name the first time you use an acronym, i.e. `Fully Qualified Name (FQN)` first and then use just `FQN` after that.